tensorflow/compiler/xla/service/all_reduce_reassociate.cc

```diff
@@ -77,17 +143,58 @@ StatusOr<bool> AllReduceReassociate::Run(
       }
 
       // Found pattern op(ar(x), ar(y)). Transform it into ar(op(x,y)).
-      HloInstruction *new_op = computation->AddInstruction(
-          inst->CloneWithNewOperands(inst->shape(), {ar0->mutable_operand(0),
-                                                     ar1->mutable_operand(0)}));
+      HloInstruction *new_op_operand0 = ar0->mutable_operand(0);
+      HloInstruction *new_op_operand1 = ar1->mutable_operand(0);
+      ;
+      if (should_promote_ar) {
+        if (convert0) {
+          auto ar0_operand = ar0->mutable_operand(0);
```

Comment:
Prefer to use explicit types when possible.

Reply:
updated